src/render/terminal.cr

```diff
@@ -55,29 +64,29 @@ module Mint
           part = ""
 
           loop do
-            char = contents[index]?.try(&.to_s)
+            char = contents[index]?
 
-            if ((char && char =~ /\s|\n|\r/) || !char) && part.size > 0
+            if ((char && char.whitespace?) || !char) && part.size > 0
               if @cursor > @width
                 @cursor = part.size
-                print "\n"
+                puts
               end
 
-              print (yield part).to_s
+              print yield part
               part = ""
             end
 
             break unless char
 
             case char
-            when "\n", "\r"
+            when '\n', '\r'
               @cursor = 0
               print char
-            when .=~(/\s/)
-              @cursor += char.size
+            when .whitespace?
+              @cursor += 1
               print char
             else
-              @cursor += char.size
+              @cursor += 1
```

Comment:
What about 2-byte characters? UTF-16

Reply:
This was done in order to keep the logic from previous implementation.

Reply:
I think it should be size since it returns the number of Unicode codepoints https://crystal-lang.org/api/0.33.0/String.html#size-instance-method

Reply:
@gdotdesign _Unicode codepoints_ in this case is equals to a character - see closely the included example code in the docs - it shows you the amount of _characters_, not bytes.

also, it's not a `String` anymore, it's a `Char`, so this method doesn't apply here.

Reply:
All right, I have run into some multiple unicode code point characters at some point but I can't remember what it was so probably not that important.

And I also vaguely remember that I needed to cast the character to a string for some reason, but if this works correctly I'm all for it :slightly_smiling_face: